slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -329,6 +330,111 @@ void SLAMServiceImpl::SetUpMapBuilder() {
 }
 
 std::string SLAMServiceImpl::GetLatestJpegMapString(bool add_pose_marker) {
+    try {
+        cartographer::io::PaintSubmapSlicesResult painted_slices =
+            GetLatestPaintedMapSlices();
+    } catch (std::exception &e) {
+        // jpeg will be empty which captures the same error
+        return "";
+    }
+    cartographer::io::PaintSubmapSlicesResult painted_slices =
+        GetLatestPaintedMapSlices();
```

Comment:
This is getting called twice. I think you meant to declare `painted_slices` outside of the try/catch, and then set it inside the try/catch.

Reply:
@nicksanford update on this. Been trying to fix this but I cannot get a PaintSubmapSLiceResult to initialize outside of the try/catch. Still messing around to try getting it to work but may have to find another way to handle errors here.

Reply:
Want to pair on this today @JohnN193 ?